dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckService.java

```diff
@@ -51,7 +51,7 @@
      * @param logger
      * @return
      */
-    boolean operationPermissionCheck(Object authorizationType, Integer userId, String permissionKey, Logger logger);
+    boolean operationPermissionCheck(Object authorizationType, Object[] projectIds, Integer userId, String permissionKey, Logger logger);
```

Comment:
I didn't see the whole code, but can we use a class to wrap the parameter, like `PermisionFactor` or anyother.
I just worry once we need to modify the parameter in the future, we need to update all method execution.

Reply:
This is a good idea. There is an intermediate layer responsible for the assembly of parameters when the business logic calls this method. In fact, there is no need to change the business logic, but the workload is concentrated in the Mock.when of the unit test.

Afterwards, a Base will be extracted for the permission Mock.when involved in the unit test.

Reply:
It would be great if this can be refactored in the future when you have time.